web/src/components/PasswordMeter.tsx

```diff
@@ -0,0 +1,136 @@
+    requireNumber: boolean;
+    requireSpecial: boolean;
+}
+
+const PasswordMeter = function (props: Props) {
+    const [progressColor] = useState(["#D32F2F", "#FF5722", "#FFEB3B", "#AFB42B", "#62D32F"]);
+    const [passwordScore, setPasswordScore] = useState(0);
+    const [maxScores, setMaxScores] = useState(0);
+    const [feedback, setFeedback] = useState("");
+    const { t: translate } = useTranslation("Portal");
+    const style = makeStyles((theme) => ({
+        progressBar: {
+            height: "5px",
+            marginTop: "2px",
+            backgroundColor: "red",
+            width: "50%",
+            transition: "width .5s linear",
+        },
+    }))();
+
+    useEffect(() => {
+        const password = props.value;
+        if (props.mode === "classic") {
+            //use mode mode
+            setMaxScores(4);
+            if (password.length < props.minLength) {
+                setPasswordScore(0);
+                setFeedback(translate("Two short, minimun length is {{len}} letters", { len: props.minLength }));
+                return;
+            }
+            if (password.length > props.maxLength) {
+                setPasswordScore(0);
+                setFeedback(translate("Two long, maximum length is {{len}} letters", { len: props.maxLength }));
+                return;
+            }
+            setFeedback("");
+            let score = 1;
+            let required = 0;
+            let hits = 0;
+            let warning = "";
+            if (props.requireLowerCase) {
+                required++;
+                const hasLowercase = /[a-z]/.test(password);
+                if (hasLowercase) {
+                    hits++;
+                } else {
+                    warning += "* " + translate("Add some lowercase letter") + "\n";
+                }
+            }
+
+            if (props.requireUpperCase) {
+                required++;
+                const hasUppercase = /[A-Z]/.test(password);
+                if (hasUppercase) {
+                    hits++;
+                } else {
+                    warning += "* " + translate("Add some UPPERCASE letter") + "\n";
+                }
+            }
+
+            if (props.requireNumber) {
+                required++;
+                const hasNumber = /[0-9]/.test(password);
+                if (hasNumber) {
+                    hits++;
+                } else {
+                    warning += "* " + translate("Add some Numbers") + "\n";
+                }
+            }
+
+            if (props.requireSpecial) {
+                required++;
+                const hasSpecial = /[^0-9\w]/i.test(password);
+                if (hasSpecial) {
+                    hits++;
+                } else {
+                    warning += "* " + translate("Add some Special character") + "\n";
+                }
+            }
+            score += hits > 0 ? 1 : 0;
+            score += required === hits ? 1 : 0;
+            setFeedback(warning);
```

Comment:
```suggestion
            setFeedback(translate("The password does not meet the password policy") + ":\n" + warning);
```

See my other suggestions for why.